src/esmockModule.js

```diff
@@ -15,10 +15,18 @@ const isObjOrFnRe = /^(object|function)$/;
 const isObjOrFn = o => isObjOrFnRe.test(typeof o) && o;
 const isDefaultDefined = o => isObj(o) && 'default' in o;
 
-// https://url.spec.whatwg.org/, eg, file:///C:/demo file://root/linux/path
-const pathAddProtocol = (pathFull, protocol) => (
-  (protocol || (resolvewith.iscoremodule(pathFull) ? 'node:' : 'file:///'))
-    + pathFull.replace(/^\//, ''));
+const FILE_PROTOCOL = 'file:///';
+
+// https://url.spec.whatwg.org/, eg, file:///C:/demo file:///root/linux/path
+const pathAddProtocol = (pathFull, protocol) => {
+  if (!protocol)
+    protocol = !resolvewith.iscoremodule(pathFull) ? FILE_PROTOCOL : 'node:';
+  if (protocol.includes(FILE_PROTOCOL))
+    pathFull = fs.realpathSync.native(pathFull);
+  if (process.platform === 'win32')
+    pathFull = pathFull.split(path.sep).join(path.posix.sep);
```

Comment:
This is a little redundant, unfortunately.

Reply:
I don't have a windows machine to investigate this area myself. I tried to get a friend to look at this but he's flakey and never did look at it lol